clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy

```diff
@@ -65,6 +68,19 @@ public class AzureComputeClient extends AzureBaseClient {
     ComputeManagementClient computeClient = new ComputeManagementClientImpl(tokenCredentials)
     computeClient.setSubscriptionId(this.subscriptionId)
     computeClient.setLogLevel(HttpLoggingInterceptor.Level.NONE)
+
+    // Add Azure Spinnaker telemetry capturing
+    computeClient.getClientInterceptors().add(new Interceptor() {
+      @Override
+      Response intercept(Interceptor.Chain chain) throws IOException {
+        Request.Builder builder = chain.request().newBuilder()
+        def oldHeaderValue = chain.request().header("User-Agent")
+        def userAgentValue = oldHeaderValue ? "AzureSpinnaker ${oldHeaderValue}" :"AzureSpinnaker"
```

Comment:
Most of the other providers are relying on the `clouddriverUserAgentApplicationName` value to be injected.
I believe you already have this value injected into `AzureNamedAccountCredentials`. Maybe you can just propagate that from `AzureNamedAccountCredentials` to `AzureCredentials` and from there to each of the `*Client.initialize*()` methods?

It would then include the version number instead of just being `AzureSpinnaker`.

It is initialized here: https://github.com/spinnaker/clouddriver/commit/45ed523c0e7340b63016f0057d9bda7ede193800#diff-98d371d4f83cb12b8688e2278bfc407cR80